inspirehep/dojson/hep/model.py

```diff
@@ -22,8 +22,16 @@
 
 """HEP model definition."""
 
+import types
 from dojson import Overdo
-from dojson import utils
+
+
+def custom_do(self, blob):
+    """Custom do function that allows extra post-processing."""
+    record = self.do(blob)
+    return record
 
 hep = Overdo()
+hep.custom_do = types.MethodType(custom_do, hep)
```

Comment:
I think you should be able to do similar to a decorator:
```python
hep._do = hep.do
hep.do = types.MethodType(custom_do, hep)
```

and in `custom_do()` you use `self._do`.